velox/exec/VectorHasher.cpp

```diff
@@ -247,36 +246,42 @@ bool VectorHasher::makeValueIdsDecoded(
   auto values = decoded_.data<T>();
 
   bool success = true;
-  rows.applyToSelected([&](vector_size_t row) INLINE_LAMBDA {
+  int numCachedHashes = 0;
+  rows.testSelected([&](vector_size_t row) INLINE_LAMBDA {
     if constexpr (mayHaveNulls) {
       if (decoded_.isNullAt(row)) {
         if (multiplier_ == 1) {
           result[row] = 0;
         }
-        return;
+        return true;
       }
     }
+
     auto baseIndex = indices[row];
-    uint64_t id = cachedHashes_[baseIndex];
-    if (id == 0) {
-      T value = values[baseIndex];
-
-      if (!success) {
-        // If all were not mappable we just analyze the remaining so we can
-        // decide the hash mode.
-        analyzeValue(value);
-        return;
+    uint64_t& id = cachedHashes_[baseIndex];
+
+    if (success) {
+      if (id == 0) {
+        T value = values[baseIndex];
+        id = valueId(value);
+        numCachedHashes++;
+        if (id == kUnmappable) {
+          analyzeValue(value);
+          success = false;
+        }
       }
-      id = valueId(value);
-      if (id == kUnmappable) {
-        analyzeValue(value);
-        success = false;
-        return;
+      result[row] = multiplier_ == 1 ? id : result[row] + multiplier_ * id;
```

Comment:
Hi @shiyu-bytedance @mbasmanova , I'm reading this and wonder is it OK to set result[row] if id is kUnmappable? Seems this changes behavior because previously result will not be set in such case.